app/controllers/RaceGroupController.java

```diff
@@ -0,0 +1,112 @@
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRaceGroup(long id) {
+        return raceGroupRepository.getRaceGroupById(id).thenApplyAsync(raceGroup -> ok(toJson(raceGroup))).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "NoResultException":
+                    res = badRequest("No racegroup with id: " + id + " is available in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> deleteAllRaceGroups() {
+        return raceGroupRepository.deleteAllRaceGroups().thenApply(raceGroups -> ok(toJson(raceGroups.collect(Collectors.toList())) + "have been deleted")).exceptionally(ex -> internalServerError(ex.getMessage()));
+    }
+
+    public CompletionStage<Result> deleteRaceGroup(long id) {
+        return raceGroupRepository.deleteRaceGroupById(id).thenApplyAsync(racegroup -> ok(toJson(racegroup) + "has been deleted")).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()) {
+                case "NoResultException":
+                    res = badRequest("Racegroup with id" + id + " not found in DB");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    @BodyParser.Of(BodyParser.Json.class)
+    public CompletionStage<Result> addRaceGroup() {
+        JsonNode json = request().body().asJson();
+        return parseRaceGroup(json).thenApply(raceGroupRepository::addRaceGroup).thenApply(raceGroup -> ok(toJson(raceGroup) + " has been added")).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "NullPointerException":
+                    res = badRequest("json format of racegroup was wrong");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    private CompletableFuture<RaceGroup> parseRaceGroup (JsonNode json) {
+        CompletableFuture<RaceGroup> completableFuture = new CompletableFuture<>();
+
+        Executors.newCachedThreadPool().submit(() -> {
+            try {
+                RaceGroup raceGroup = new RaceGroup();
+                raceGroup.setTimestamp(Timestamp.valueOf(json.findPath("timestamp").textValue()));
+                String raceGroupType = json.findPath("type").textValue();
+                raceGroup.setRaceGroupType(RaceGroupType.valueOf(raceGroupType));
+                raceGroup.setHistoryGapTime(json.findPath("actualGapTime").longValue());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "actualGapTime" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)